Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -169,6 +169,8 @@ class TopSitesPanel: UIViewController {
     }
 
     private func refreshHistory(frequencyLimit: Int) {
+        // We double the requested limit in order to generate some leeway for "grouping by domain". Hopefully
+        // that returns enough entries that we can group but still have frequencyLimit results.
```

Comment:
I think there are two better ways to do this.

1. Pass the limit into a particular subquery, if we can arrange it that way.
2. Push the grouping logic down into SQL, so you get the right rows directly and need to do no grouping in memory at all.

The latter should be eminently possible; they're already sorted by frecency and grouped by domain, so you just want to take the greatest-n-per-group solution where n=1.